### PR TITLE
Fix Depwarns

### PR DESCRIPTION
**Pull request by SimonDanisch**
I just tried using WinRPM on 0.4, not sure if that's a sufficient test... Actually, the depwarns are so bad, that WinRPM just stops installing binaries without any error. So this also needs the depwarn removal from LibExpat (amitmurthy/LibExpat.jl#45)

### CODE REVIEW

**Comment by timholy** [comment]
LGTM. OK to merge?

**Comment by timholy** [comment]
It does seem rather important to test this on AppVeyor :smile:. I'll leave others to it.

**Comment by tkelman** [comment]
Oh and Julia 0.3 is apparently still supported here too but that doesn't need to remain the case.

File: src/WinRPM.jl [hunk 1]
```diff
@@ -349,7 +350,7 @@ end
 
 install(pkg::AbstractString, arch::AbstractString=OS_ARCH; yes = false) = install(select(lookup(pkg, arch),pkg); yes = yes)
 
-function install(pkgs::Vector{ASCIIString}, arch = OS_ARCH; yes = false)
+function install(pkgs::Vector{String}, arch = OS_ARCH; yes = false)
```
Comment (tkelman) [inline src/WinRPM.jl]: I don't think this will work on 0.4 or earlier, `Compat.ASCIIString` is less risky

**Comment by SimonDanisch** [comment]
Okay, removed support for 0.3 and squashed...
Tested on 0.4 and 0.5 with:
```
using WinRPM
WinRPM.update()
WinRPM.install("gcc")
```
Would it make sense to add this as a minimal test for appveyor and travis?

File: src/WinRPM.jl [hunk 2]
```diff
@@ -1,17 +1,23 @@
 module WinRPM
 
-@unix_only using HTTPClient.HTTPC
 using Compat
+import Compat.String
+
+if is_unix()
+  using HTTPClient.HTTPC
```
Comment (tkelman) [inline src/WinRPM.jl]: 4 space indent

**Comment by tkelman** [comment]
yeah something simple would be better than nothing. maybe a few bindeps integration tests via `test/REQUIRE` would also be good

**Comment by tkelman** [comment]
needs rebase

**Comment by SimonDanisch** [comment]
Allrighty! Hope this is ready now :D

File: src/WinRPM.jl [hunk 3]
```diff
@@ -42,8 +48,13 @@ function __init__()
     update(false, false)
 end
 
-@unix_only download(source::AbstractString) = (x=HTTPC.get(source); (bytestring(x.body),x.http_code))
-@windows_only function download(source::AbstractString; retry = 5)
+if is_unix()
+  function download(source::AbstractString)
```
Comment (tkelman) [inline src/WinRPM.jl]: 4-space please

File: src/WinRPM.jl [hunk 4]
```diff
@@ -1,18 +1,24 @@
...
 using LegacyStrings
 
+
 import Base: show, getindex, wait_close, pipeline_error
 
 #export update, whatprovides, search, lookup, install, deps, help
 
-if OS_NAME == :Windows
-    const OS_ARCH = WORD_SIZE == 64 ? "mingw64" : "mingw32"
+if is_windows()
```
Comment (tkelman) [inline src/WinRPM.jl]: minimum Compat version probably needs increasing

File: src/WinRPM.jl [hunk 5]
```diff
@@ -455,15 +469,17 @@ function do_install(package::Package)
...
             wait_close(out)
             println(stdoutstr)
             err = pc
-            @unix_only cd(installdir) do
-                if success(`rpm2cpio $path2` | `cpio -imud`)
-                    err = nothing
-                end
+            if is_unix()
+                cd(installdir) do
+                  if success(`rpm2cpio $path2` | `cpio -imud`)
```
Comment (tkelman) [inline src/WinRPM.jl]: here as well

Comment (SimonDanisch) [inline src/WinRPM.jl]: ugh, my editor settings are messed up on windows -.-

Comment (tkelman) [inline src/WinRPM.jl]: if your editor is Juno, the mismatch of default indent there relative to the standard julia style is a continued bug that users of that editor should continue reporting to its author

File: src/WinRPM.jl (hunk 4 above)
Comment (SimonDanisch) [inline src/WinRPM.jl]: `is_windows` seems to be in 0.7.16, from what I can tell!

Comment (tkelman) [inline src/WinRPM.jl]: 0.7.20 https://github.com/JuliaLang/Compat.jl/commit/85f886227c332507ec5b31627e0b3a9fbb6aa274

File: src/WinRPM.jl [hunk 6]
```diff
@@ -42,25 +48,33 @@ function __init__()
...
-        else
-            warn("Unknown download failure, error code: $res")
+            warn("Retry $i/$retry downloading: $source")
         end
-        warn("Retry $i/$retry downloading: $source")
+        return "", 0
     end
-    return "",0
+else
+    error("Platform not supported: $(Sys.KERNEL)")
```
Comment (tkelman) [inline src/WinRPM.jl]: `Compat.KERNEL`, otherwise won't work on 0.4 (not that this line is likely)

File: src/WinRPM.jl [hunk 7]
```diff
@@ -455,15 +469,17 @@ function do_install(package::Package)
...
-            @unix_only cd(installdir) do
-                if success(`rpm2cpio $path2` | `cpio -imud`)
-                    err = nothing
-                end
+            if is_unix()
+                cd(installdir) do
+                    if success(`rpm2cpio $path2` | `cpio -imud`)
+                        err = nothing
+                    end
+              end
```
Comment (tkelman) [inline src/WinRPM.jl]: if you want to give me commit access to your fork I can also fix these

**Comment by tkelman** [comment]
Thanks for bearing with my pickiness. I'm indirectly testing this at https://ci.appveyor.com/project/tkelman/lightxml-jl/build/1.0.22

**Comment by SimonDanisch** [comment]
Thanks Tony! Sorry for these messy fixes inbetween trains ;)

**Comment by tkelman** [comment]
I'll take whatever help I can get with this stuff!

**Comment by tkelman** [comment]
There's still a method overwrite warning, but looks like this works.

**Pull request merged**